src/main/java/com/innovatrics/mrz/types/MrzDate.java

```diff
@@ -103,4 +120,12 @@ public int hashCode() {
     public int compareTo(MrzDate o) {
         return Integer.valueOf(year * 10000 + month * 100 + day).compareTo(o.year * 10000 + o.month * 100 + o.day);
     }
+
+    /**
+     * Returns the date validity
+     * @return Returns a boolean true if the parsed date is valid, false otherwise
+     */
+    public boolean isValidDate() {
```

Comment:
I think it should be checked during parse so in case of invalid date, it can be also flagged in MrzRecord